tests/cpp/message_queue_test.cc

```diff
@@ -101,5 +101,8 @@ TEST(MessageQueueTest, MultiThread) {
   for (int i = 0; i < kNumOfProducer; ++i) {
     thread_pool[i]->join();
   }
+  for (int i = 0; i < kNumOfProducer; ++i) {
+    delete thread_pool[i];
```

Comment:
how about utilizing move semantic directly instead of `new` operator in line 94, then we don't need delete explicitly.

Reply:
done